src/main/java/ToDoList.java

```diff
@@ -0,0 +1,133 @@
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Scanner;
+import java.nio.file.Path;
+import java.util.regex.Pattern;
+
+
+public class ToDoList {
+    private ArrayList<Task> arr = new ArrayList<>(); //1-based index
```

Comment:
Generally, even for user inputs that start with 1-index. You should use 0-index for your internal data structures and convert the user inputs index. Otherwise you'll have to do a lot of necessary steps down the road with your methods such as deleting, checking for max length, and your constructor having to add a dummy task which takes up resources.